### ISSUE TITLE
Question about pre-text reconstruciton

### ISSUE THREAD

**Issue by junseokoh1** [XiaoyuShi97/FlowFormerPlusPlus#7]
Hi, thank you for your nice work and paper.
I'm quite interested in you work and I have some question..

In section 3.4,  paper said " $c_x$, which is aggregated cost feature, should be similar to that of the local cost patch $q_x$ ".
I'm hard to understand why the center of $c_x$ should be similar with $q_x$.

Could you explain about this?

And I guess in pre-training, model didn't learn to reconstruct the masked regions of cost memory is it right? (Instead predict larger patch of cost volume)

I will be appreciated for your reply, thank you very much!
Best wishes

**Comment by XiaoyuShi97** [comment]
Hi, thanks for your interest in our work.

1) As FlowFormer follows RAFT to use to recurrent updating schema, in each iteration, it retrieves cost patch using current estimated location as center. To ensure pretraining-finetuning consistency, we hope the cost feature extracted from the cost memory to be aware of the position information and with larger receptive field. So c_x should focus on the area centered at current estimated location. In other words, c_x is changing as the recurrent process goes rather than a global max similarity.
2) It achieves both goals. As the input cost volume is partially masked, it learns to reconstruct the unvisible parts. Meanwhile, as mentioned in 1), it learning to reconstruct specific areas with the "prompt" of q_x^s.

Let me know if you have other questions.

**Comment by junseokoh1** [comment]
Hi, thanks for your kind reply.

About question 2,
As I understand it, when randomly sample small cost patch q_x^s, q_x^s can have masked region as the input cost volume is already partially masked. So it can achieves both goals. Is it right?

**Comment by XiaoyuShi97** [comment]
Hi. We extract clean q_x^s from original unmasked cost volume to keep aligned with finetuning. But in implementation, the same area of q_x^s in the masked cost volume can have mask.
